extension/test/integration/refund-payment.handler.spec.js

```diff
@@ -0,0 +1,103 @@
+const { expect } = require('chai')
+
+const iTSetUp = require('./integration-test-set-up')
+const ctpClientBuilder = require('../../src/ctp')
+const {
+  CTP_ADYEN_INTEGRATION, CTP_INTERACTION_TYPE_REFUND, CTP_PAYMENT_CUSTOM_TYPE_KEY
+} = require('../../src/config/constants')
+const { createAddTransactionAction } = require('../../src/paymentHandler/payment-utils')
+
+describe('::refund::', () => {
+  let ctpClient
+  let payment
+
+  beforeEach(async () => {
+    ctpClient = ctpClientBuilder.get()
+    await iTSetUp.cleanupCtpResources(ctpClient)
+    await iTSetUp.initServerAndExtension({ ctpClient })
+    const paymentDraft = {
+      amountPlanned: {
+        currencyCode: 'EUR',
+        centAmount: 1000,
+      },
+      paymentMethodInfo: {
+        paymentInterface: CTP_ADYEN_INTEGRATION
+      },
+      custom: {
+        type: {
+          typeId: 'type',
+          key: CTP_PAYMENT_CUSTOM_TYPE_KEY
+        },
+        fields: {}
+      },
+      transactions: [
+        {
+          type: 'Charge',
+          amount: {
+            currencyCode: 'EUR',
+            centAmount: 1000,
+          },
+          interactionId: '883592826488441K',
+          state: 'Success'
+        }
+      ]
+    }
+
+    const result = await ctpClient.create(ctpClient.builder.payments, paymentDraft)
+    payment = result.body
+  })
+
+  afterEach(async () => {
+    await iTSetUp.stopRunningServers()
+    await iTSetUp.cleanupCtpResources(ctpClient)
+  })
+
+  it('given a payment with "charge success transaction" ' +
```

Comment:
can we also add a test which trying to refund more the payment value?

Reply:
It doesn't make sense here, there's no difference if you refund more or not. We don't check it, it's up to Adyen to check.